python/mxnet/ndarray/ndarray.py

```diff
@@ -968,6 +973,14 @@ def reshape(self, shape):
         array([[-1., -1., -1.],
                [-1., -1., -1.]], dtype=float32)
         """
+        if len(shape) == 1 and isinstance(shape[0], (list, tuple)):
+            shape = shape[0]
+        elif not len(shape):
+            for key, value in kwargs.items():
```

Comment:
no need to iterate. test for len(kwargs) == 1 and kwargs.get('shape', None) directly

Reply:
it also helps throw exception with the invalid argument name, which is consistent with numpy's behavior.